nvme.c

```diff
@@ -221,67 +217,53 @@ void *nvme_realloc(void *p, size_t len)
 	return result;
 }
 
-static void *__nvme_alloc_huge(size_t len, bool *huge)
-{
-	void *p;
-
-	if (!posix_memalign(&p, getpagesize(), len)) {
-		*huge = false;
-		memset(p, 0, len);
-		return p;
-	}
-	return NULL;
-}
-
 #define HUGE_MIN 0x80000
 
-#ifdef CONFIG_LIBHUGETLBFS
-void nvme_free_huge(void *p, bool huge)
+void nvme_free_huge(void *p, size_t len)
 {
-	if (huge) {
-		if (p)
-			free_hugepage_region(p);
-	} else {
+	size_t _len = ROUND_UP(len, 0x1000);
+
+	if (_len < HUGE_MIN) {
 		free(p);
+		return;
 	}
+
+	munmap(p, _len);
 }
 
-void *nvme_alloc_huge(size_t len, bool *huge)
+void *nvme_alloc_huge(size_t len)
```

Comment:
This function name is also same as for removeing `huge` from the name.

Reply:
If I understood you correctly, you would like to see both allocation function combined into one? 

Not really thought about it at so far. Surely, it's possible but I would like to defer this discussion for now.  Let's get the library dependency removed first.

Reply:
I see. Yes your understainding correct and right as not a fix for the issue.